specs/eventing/overview.md

```diff
@@ -1,97 +1,88 @@
+### Trigger
 
-![Resource Types Overview](images/resource-types-overview.svg)
+**Trigger** defines a filtered delivery option to extract events delivered to a
+**Broker** and route them to an **Addressable** destination. Trigger implements
+uniform event filtering based on the CloudEvents attributes associated with the
+event, ignoring the payload (which might be large and/or binary and need not be
+parsed during event routing). The addressable interface contract allows Triggers
+to deliver events to a variety of different destinations, including external
+resources such as a virtual machine or SaaS service.
 
-Channels as well as Sources are defined by independent CRDs that can be
-installed into a cluster. Both Sources and Channels implementations can be
-created directly. A Channel however offers also a way to create the backing
-implementation as part of the generic Channel (using a _channelTemplate_).
+## Messaging
 
-## Trigger
+### Channel
 
-**Trigger** describes a registration of interest on a filter set of events
-delivered to a _Broker_ which should be delivered to an _Addressable_. Events
-selected by a _Trigger_ are buffered independently from other _Triggers_, even
-if they deliver to the same _Addressable_.
+**Channel** provides an abstract interface which may be fulfiled by several
+concrete implementations of a backing asynchronous fan-out queue. The common
+abstraction provided by channel allows both the composition of higher-level
+constructs for chained or parallel processing of events, and the replacement of
+particular messaging technologies (for example, allowing a development
+environment to use a lower-reliability channel compared with the production
+environment).
 
-For more details, see [Kind: Trigger](spec.md#kind-trigger).
+### Subscription
 
-## Broker
+**Subscription** defines a delivery destination for all events sent to a
+**Channel**. Events sent to a channel are delivered to _each_ subscription
+_independently_ -- a subscription maintains its own list of undelivered events
+and will manage retry indpendently of any other subscriptions to the same
+channel. Like **Trigger**, subscriptions use the **Addressable** interface
+contract to support event delivery to many different destination types.
 
-**Broker** provides an eventing mesh. This allows producers to deliver events to
-a single endpoint and not need to worry about the routing details for individual
-consumers.
+## Interface Contracts
 
-For more details, see [Kind: Broker](spec.md#kind-broker).
+In addition to the concrete types described above in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the other resources must support. The following
+interface contracts define a set of expected resource fields on an referenced
+resource.
 
-## Subscription
+### Addressable
```

Comment:
Yeah, I think in this section we might want to add a Destination in the sense that it's how you specif a Ref to a resource that then must support Addressable (plus special cased k8s service).

Reply:
well, unless of course you use the URI field, which then provides the mechanism to point to external sinks (like your examples above about VMs, SaaS, etc.)

Reply:
Your point about Destination being a tagged union and it being helpful to be able to reference it elsewhere is a good point; I've added Destination as an interface / object fragment. I don't think it's a partial schema/duck type, because Destination doesn't say anything about _where_ in the object it occurs.

Reply:
In particular, Destination allows pointing at a lot of different things, so it's worth having a section explaining how that works.